Scenes/UserInterface/UIContainer.gd

```diff
@@ -48,6 +50,7 @@ func _on_RevivalEnterBtn_pressed():
 
 func _on_TavernEnterBtn_pressed():
 	revival_enter_btn.set_pressed(false)
+	music_manager.play_music(music_manager.Musics.TAVERN_MUSIC)
```

Comment:
Player can enter the tavern by dying, what means that player won't hear proper track. Please use this method in other place or copy it so it includes the other case

Reply:
Another things is that players can click this button and music starts playing before they enter the tavern

Reply:
Ok, I'm working on it, I just added a manager and plugged in a few places ;)